### PR TITLE
TNF handler generation tool improvements.

### PR DESCRIPTION
**Pull request by greyerof**
Fixed so the tnf tool can be launched anywhere. If no envar
TNF_HANDLER_SRC detected, the tool assumes you're in the repo root
folder so the handlers folder should be at ./pkg/tnf/handlers.

+ Minor refactors.

### CODE REVIEW

File: cmd/tnf/main.go [hunk 1]
```diff
@@ -36,53 +38,97 @@ var (
...
+	}
+
+	return handlersDirectory, nil
+}
+
+func generateHandlerFilesFromTemplates(handlerTemplatesDirectory, newHandlerDirectory string, myhandler myHandler) error {
+	templateFilePath := path.Join(handlerTemplatesDirectory, "doc.tmpl")
+	renderedFileName := docFileName
+
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
```
Comment (hamadise) [inline cmd/tnf/main.go:74]: can you add a log.Error here.
Also, I think this is worth a panic, we can't do anything if the files can't be created

File: pkg/tnf/handlers/handler_template/handler.tmpl [hunk 2]
```diff
@@ -35,17 +35,27 @@ const (
...
+	}
+}
+
+// Args returns the initial execution/send command strings for handler {{ .UpperHandlername }}.
 func (h *{{ .UpperHandlername }}) Args() []string {
 	return h.args
 }
 
 // GetIdentifier returns the tnf.Test specific identifier.
 func (h *{{ .UpperHandlername }}) GetIdentifier() {
```
Comment (hamadise) [inline pkg/tnf/handlers/handler_template/handler.tmpl:54]: can you fix the signature of the function and return a nil. is it allowed ?

File: pkg/tnf/handlers/handler_template/handler.tmpl [hunk 3]
```diff
@@ -55,28 +65,28 @@ func (h *{{ .UpperHandlername }}) Result() int {
...
-	// TODO : add the matching logic in here
 }
 
-// ReelTimeout does nothing, {{ .UpperHandlername }} requires no explicit intervention for a timeout.
+// ReelTimeout function for {{ .UpperHandlername }} will be called by the reel FSM when a expect timeout occurs.
 func (h *{{ .UpperHandlername }}) ReelTimeout() *reel.Step {
+	// TODO : Add code here in case a timeout reaction is needed.
 	return nil
-	// TODO : fill the stub
 }
```
Comment (hamadise) [inline pkg/tnf/handlers/handler_template/handler.tmpl:88]: general question, does the code compile properly after generation ?

File: cmd/tnf/main.go [hunk 4]
```diff
@@ -36,53 +38,97 @@ var (
...
 
-	pathfile := path.Join(handlerDirectory, "handler_template", "doc.tmpl")
-	namefile := "" + "doc.go"
-	err = createfile(pathfile, namefile, myhandler, newHandlerDirectory) // here creating file by doc.tmpl
-	if err != nil {
+	templateFilePath = path.Join(handlerTemplatesDirectory, "handler.tmpl")
+	renderedFileName = myhandler.LowerHandlername + ".go"
+
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
 		return err
```
Comment (hamadise) [inline cmd/tnf/main.go:89]: you probably can have a loop like this!
```
for s:= range []string{handler_test,handler.temp,doc.template}{
 // read string
 // replace handler with real_hander and temp with go
 // generate file
}
```

File: cmd/tnf/main.go (hunk 1 above)
Comment (greyerof) [reply cmd/tnf/main.go]: I've added an error log, but I don't think we need to panic here. If this function fails, the error is catched and the program returns immediately with a graceful exit.

File: pkg/tnf/handlers/handler_template/handler.tmpl (hunk 2 above)
Comment (greyerof) [reply pkg/tnf/handlers/handler_template/handler.tmpl]: Done. Surprisingly, the previous "return nil" was successfully compiled.

File: cmd/tnf/main.go (hunk 4 above)
Comment (greyerof) [reply cmd/tnf/main.go]: Not sure if my new approach is exactly what you were suggesting, but at least it avoids the previous repetitive code.

File: pkg/tnf/handlers/handler_template/handler.tmpl (hunk 3 above)
Comment (greyerof) [reply pkg/tnf/handlers/handler_template/handler.tmpl:90]: Yes, somehow it worked.

**Review by hamadise — APPROVED** [review]

File: cmd/tnf/main.go (hunk 4 above)
Comment (hamadise) [reply cmd/tnf/main.go]: it's not the same, but it's good!

File: DEVELOPING.md [hunk 5]
```diff
@@ -606,11 +606,14 @@ oc get pod %s -n %s -o go-template='{{len .spec.containers}}{{"\n"}}'
 
 ## Adding new handler
 
-To facilitate adding new handlers, the developers can leverage existing infrastructure to generate repetitive code.
-As an example, to generate handler with name Xyz, the command generate handler Xyz can be used(we need to run this command 
-in the folder "cmd/tnf/" and we run the command directly here).
+To facilitate adding new handlers, the "tnf" utility has been created to help developers to avoid generating repetitive code.
+As an example, to generate a new handler with named MyHandler, the command "tnf generate handler MyHandler" can be used.
```
Comment (edcdavid) [inline DEVELOPING.md:610]: Add instructions to build or run the cmd/tnf/main.go. Use hyperlinks to reference files that are in the repo

**Review by sebrandon1 — APPROVED** [review]

File: DEVELOPING.md [hunk 6]
```diff
@@ -606,11 +606,14 @@ oc get pod %s -n %s -o go-template='{{len .spec.containers}}{{"\n"}}'
 
 ## Adding new handler
 
-To facilitate adding new handlers, the developers can leverage existing infrastructure to generate repetitive code.
-As an example, to generate handler with name Xyz, the command generate handler Xyz can be used(we need to run this command 
-in the folder "cmd/tnf/" and we run the command directly here).
+To facilitate adding new handlers, the "tnf" utility has been created to help developers to avoid generating repetitive code.
```
Comment (jun-rh) [inline DEVELOPING.md:609]: generating -> writing

**Review by jun-rh — APPROVED** [review]

File: DEVELOPING.md (hunk 5 above)
Comment (greyerof) [reply DEVELOPING.md]: Done.

File: DEVELOPING.md (hunk 6 above)
Comment (greyerof) [reply DEVELOPING.md]: Done.

File: DEVELOPING.md [hunk 7]
```diff
@@ -606,11 +606,22 @@ oc get pod %s -n %s -o go-template='{{len .spec.containers}}{{"\n"}}'
 
 ## Adding new handler
 
-To facilitate adding new handlers, the developers can leverage existing infrastructure to generate repetitive code.
-As an example, to generate handler with name Xyz, the command generate handler Xyz can be used(we need to run this command 
-in the folder "cmd/tnf/" and we run the command directly here).
+To facilitate adding new handlers, the "tnf" utility has been created to help developers to avoid writing repetitive code. The tnf tool [source code is here](https://github.com/test-network-function/test-network-function/tree/main/cmd/tnf) and can be built with the following command:
```
Comment (jun-rh) [inline DEVELOPING.md:609]: "and" should be removed?

**Comment by hamadise** [comment]
@greyerof can you rebase your PR ?

**Review by edcdavid — APPROVED** [review]

**Pull request merged**